File: Class 3/mwc15c5p3.java
import java.util.*;
import java.io.*;

public class mwc15c5p3{
  public static void main(String[] args) throws IOException{
    BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    
    String[] t = in.readLine().split(" ");
    int N = Integer.parseInt(t[0]), M = Integer.parseInt(t[1]);
    
    HashSet<String> hs = new HashSet<String>();
    HashSet<String> ans = new HashSet<String>();
    
    t = in.readLine().split(" ");
    for(int i = 0; i < N; i++){
      hs.add(t[i]);
    }
    
    t = in.readLine().split(" ");
    for(int i = 0; i < M; i++){
      if(hs.contains(t[i]))
        ans.add(t[i]);
    }
    
    System.out.println(ans.size());
  }
}

File: Class 3/mwc15c5p3.py
from sys import stdin

line = stdin.readline().strip()
N, M = line.split()

N = int(N)
M = int(M)

line = stdin.readline().strip()
words = line.split()

hs = set()
ans = set()

for word in words:
    hs.add(word)

line = stdin.readline().strip()
words2 = line.split()

for word in words2:
    if word in hs:
        ans.add(word)

print(len(ans))
